src/SignalR/server/Core/src/Internal/DefaultHubDispatcher.cs

```diff
@@ -389,7 +389,7 @@ async Task ExecuteInvocation()
                 }
                 finally
                 {
-                    (enumerator as IDisposable)?.Dispose();
+                    await enumerator.DisposeAsync();
```

Comment:
Change this code as well. Can we just do await foreach here?

Reply:
If we do this, we'll have to always wrap the IAsyncEnumerable even if T is a reference type, so we can flow the CancellationToken through to underlying GetAsyncEnumerator implementation. `await foreach` always passes in a default token.